src/main/java/org/jenkinsci/plugins/workflow/cps/Snippetizer.java

```diff
@@ -349,17 +350,69 @@ private static boolean isSingleMap(Map<String, ?> args) {
         return Jenkins.getActiveInstance().getDescriptorByName(id);
     }
 
-    @Restricted(DoNotUse.class)
-    public Collection<? extends StepDescriptor> getStepDescriptors(boolean advanced) {
-        TreeSet<StepDescriptor> t = new TreeSet<StepDescriptor>(new StepDescriptorComparator());
+    @Restricted(NoExternalUse.class)
+    public Collection<QuasiDescriptor> getQuasiDescriptors(boolean advanced) {
+        TreeSet<QuasiDescriptor> t = new TreeSet<>();
         for (StepDescriptor d : StepDescriptor.all()) {
             if (d.isAdvanced() == advanced) {
-                t.add(d);
+                t.add(new QuasiDescriptor(d));
+                if (d.isMetaStep()) {
+                    DescribableModel<?> m = new DescribableModel<>(d.clazz);
+                    Collection<DescribableParameter> parameters = m.getParameters();
+                    if (parameters.size() == 1) {
+                        DescribableParameter delegate = parameters.iterator().next();
+                        if (delegate.isRequired()) {
+                            if (delegate.getType() instanceof HeterogeneousObjectType) {
+                                // TODO HeterogeneousObjectType does not yet expose symbol information, and DescribableModel.symbolOf is private
+                                for (DescribableModel<?> delegateOptionSchema : ((HeterogeneousObjectType) delegate.getType()).getTypes().values()) {
+                                    Class<?> delegateOptionType = delegateOptionSchema.getType();
+                                    Descriptor<?> delegateDescriptor = Jenkins.getActiveInstance().getDescriptorOrDie(delegateOptionType.asSubclass(Describable.class));
+                                    Symbol symbol = delegateDescriptor.getClass().getAnnotation(Symbol.class);
+                                    if (symbol != null && symbol.value().length > 0) {
+                                        t.add(new QuasiDescriptor(delegateDescriptor));
+                                    }
+                                }
+                            }
+                        }
+                    } // TODO currently not handling metasteps with other parameters, either required or (like GenericSCMStep) not
```

Comment:
Why? I understand that there's no way to offer configuration for those extra parameters without further change, but isn't it still useful to list them in Snippetizer?